processing/src/main/java/io/druid/query/MetricsEmittingQueryRunner.java

```diff
@@ -117,7 +119,13 @@ public void after(boolean isDone, Throwable thrown)
             if (creationTimeNs > 0) {
               queryMetrics.reportWaitTime(startTimeNs - creationTimeNs);
             }
-            queryMetrics.emit(emitter);
+            try {
+              queryMetrics.emit(emitter);
+            }
+            catch (Exception e) {
+              // Query should not fail, because of emitter failure. Swallowing the exception.
+              log.error("Failure while trying to emit [%s] with stacktrace [%s]", emitter.toString(), e);
```

Comment:
This should be `log.error(e, "Failure emitting query metrics with emitter[%s]", emitter)`. The exception goes first, or else it won't be interpreted as a stack trace.

Reply:
you are right, sorry missed that.